### PR TITLE
Add monorepo demo example page

### PR DESCRIPTION
**Pull request by TomFern**
This only adds the monorepo demo example page with fully annotated YAML config.

### CODE REVIEW

File: docs/examples/change-based-execution-for-monorepos.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+maintains a reference demo on GitHub:
+
+- [Monorepo project on GitHub][demo-project]
+
+You’ll find a Semaphore configuration file with a sample CI pipeline at `.semaphore/semaphore.yml`
+
+The repository contains three individually deployable applications. They can be found under the `/services` folder.
+
+- **Billing**: a billing system written in Go. Uses [mux][mux] to provide an HTTP endpoint on port 8000.
+- **User**: a user account management application. Written in Ruby, it employs an in-memory database and uses [Sinatra][sinatra] to expose an HTTP endpoint.
```
Comment (AleksandarCole) [inline docs/examples/change-based-execution-for-monorepos.md:21]: @TomFern Sinatra link seems to be missing.

**Review by AleksandarCole — APPROVED** [review]
Noticed only one small bug, left a comment. 
This looks great!

**Comment by TomFern** [comment]
> Noticed only one small bug, left a comment.

Fixed it. Should be ready to merge. Thank you @AleksandarCole 😀

**Review by shiroyasha — APPROVED** [review]

**Comment by shiroyasha** [comment]
Reviewed it. Looks good from my side 👍

**Review by DamjanBecirovic — APPROVED** [review]
Looks great :+1:

**Pull request merged**